### ISSUE TITLE
Substantial difficulties in getting full Excel range back

### ISSUE THREAD

**Issue by kkkmail** [spreadsheetlab/XLParser#75]
The parser works great and produces the expression trees. However, Excel references to the cell with the sheet name, which contains space, or to excel range produce a tree. For example:
`'Sheet 1'!A1` becomes a tree with the leafs containing `'`, `Sheet 1'`, and `A1`
and the range `'Sheet 1'!A1:A5` produces the leafs like `'`, `Sheet 1'`, `A1`, `:`, and `A5`.

This is extremely inconvenient because if you are working with the real Excel file, then you need to get the real Excel references out of it and not an expression tree. 

Enclosed rough example is what I mean. 
[GetPrecedents.zip](https://github.com/spreadsheetlab/XLParser/files/2044637/GetPrecedents.zip)

I don't really like how it works because it makes a lot of assumption about how the expression trees are structured for the cases above. 

In addition, if the expression is in the brackets, like `=(IF(...))`, then the parser produces another tree level, which makes it even trickier to determine if it is a function call. See `IsExcelReference` and a part with 
```
                if
                (
                    node.ChildNodes.Count > 0 &&
                    node.ChildNodes[0].Is(GrammarNames.Reference) &&
                    node.ChildNodes[0].ChildNodes.Count > 0 &&
                    node.ChildNodes[0].ChildNodes[0].Is(GrammarNames.ReferenceFunctionCall)
                )
                {
                    //ReferenceFunctionCall like in (IF(...))
                    return false;
                }
```
In case of Excel range reference, a cell parent is needed to determine what to do with the leaf. 

PS I can repost the code once I brush it up and add unit tests.

**Comment by kkkmail** [comment]
Here is the updated solution. Place the file `GetPrcedents.xslm` from the folder `Data` into `C:\Temp\` and run the solution. Please, let me know if you need any further comments. 
[RaterParser.zip](https://github.com/spreadsheetlab/XLParser/files/2063270/RaterParser.zip)

**Comment by roelvandersteen** [comment]
I understand the problems you are having, but I see no way of changing this without breaking the existing parse tree structures. Any ideas?

**Comment by kkkmail** [comment]
@roelvandersteen I think that in general I disagree with the parsed tree structure and, therefore, I agree with you :) that it is not possible to change it without breaking the tree structure! 

I solved the problems that I needed to solve using several "workaround" methods, which covered all cases that I encountered. Perhaps, having enough of such methods will do the trick and they will "exhaust" all possible variants. After all it is possible to keep adding more and more of such exception rules to cover more and more edge cases.
